Services/DataX.Utilities/DataX.Utilities.Web/RolesCheck.cs

```diff
@@ -35,9 +64,12 @@ public static void EnsureWriter(HttpRequest request)
             }
 
             var userrole = request.Headers[Constants.UserRolesHeader];
+            var userID = request.Headers[Constants.UserIdHeader];
+            AddWhitelistedTestClientUserId();
```

Comment:
This will likely add the ID multiple time, i.e. everytime an API is called and this checks run.  I wonder if the Add.. call can be put somewhere else, in the initiliation of the service for example.

Reply:
Hashset adds unique values.